frontend/src/metabase/admin/tasks/containers/TaskModal.tsx

```diff
@@ -0,0 +1,32 @@
+import { goBack } from "react-router-redux";
+import { t } from "ttag";
+
+import { useGetTaskQuery } from "metabase/api";
+import Code from "metabase/components/Code";
+import ModalContent from "metabase/components/ModalContent";
+import { useDispatch } from "metabase/lib/redux";
+import { Center, Loader } from "metabase/ui";
+
+type TaskModalProps = {
+  params: { taskId: number };
+};
+
+export const TaskModal = ({ params }: TaskModalProps) => {
+  const dispatch = useDispatch();
+  const { data, isFetching } = useGetTaskQuery(params.taskId);
+
+  if (!data) {
+    return null;
+  }
+
+  return (
+    <ModalContent title={t`Task details`} onClose={() => dispatch(goBack())}>
+      {isFetching && (
+        <Center h="4rem">
+          <Loader data-testid="loading-spinner" size="2rem" />
+        </Center>
+      )}
```

Comment:
Entities had a support for the loading/fetching mechanism out of the box.
We need to take care of that separately with the RTK api.

Reply:
They also had error handling out of the box (with `LoadingAndErrorWrapper`) - let's make sure we don't lose it.